File: LeetCodeCPlusPlus/py66.cpp
// 加1
#include <vector>
using namespace std;
class Solution {
public:
    vector<int> plusOne(vector<int>& digits) {
        int len = digits.size();
        int add_on = 1;
        for(int i=len-1;i>=0;i--) {
            digits[i] += add_on;
            if (digits[i]<10) {
                add_on = 0;
                break;
            }
            else {
                digits[i] -= 10;
                add_on = 1;
            }
        }
         
        if (add_on == 1) {
            vector<int> ans(len, 0);
            ans.insert(ans.begin(),1);
            return ans;
        } else
            return digits;
        
    }
};

File: LeetCodePython/py66.py
# 加1
from typing import List
class Solution:
    def plusOne(self, digits: List[int]) -> List[int]:
        n = len(digits)
        add_on = 1
        for i in range(n-1, -1, -1):
            digits[i] += add_on
            
            if digits[i]<10:
                add_on = 0
                break
            else:
                digits[i] -= 10
                add_on = 1
           
        if add_on:
            result = [1]
            result.extend(digits)
            return result
        else:
            return digits
                